### ISSUE TITLE
Using Wandb Sweep with Skorch Failed: Need Guide

### ISSUE THREAD

**Issue by UTUnex** [wandb/wandb#1212]
### Description

I tried to use wandb sweep to tune the hyperparameters in skorch and perform the visulization but I didn't get any result.

### What I Did
My code is :  
```
# Load libraries
import wandb  
from skorch.callbacks import WandbLogger  
import torch  
from torch import nn  
import torch.nn.functional as F  

# Set random seed and device
torch.manual_seed(42); # 
torch.cuda.manual_seed(42) #
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

# Data Preparation 
import numpy as np
from sklearn.datasets import make_classification

X_train, y = make_classification(1000, 20, n_informative=10, random_state=42)
X_train, y = X_train.astype(np.float32), y.astype(np.int64)

X_train.shape, y.shape, y.mean()

# Net define
class ClassifierModule(nn.Module):
    def __init__(
            self,
            num_units=10,
            nonlin=F.relu,
            dropout=0.5,
    ):
        super(ClassifierModule, self).__init__()
        self.num_units = num_units
        self.nonlin = nonlin
        self.dropout = dropout

        self.dense0 = nn.Linear(20, num_units)
        self.nonlin = nonlin
        self.dropout = nn.Dropout(dropout)
        self.dense1 = nn.Linear(num_units, 10)
        self.output = nn.Linear(10, 2)

    def forward(self, X, **kwargs):
        X = self.nonlin(self.dense0(X))
        X = self.dropout(X)
        X = F.relu(self.dense1(X))
        X = F.softmax(self.output(X), dim=-1)
        return X

# Train and tune
import skorch
from skorch import NeuralNetClassifier
from skorch.callbacks import EpochScoring
f1 = EpochScoring(scoring='f1', lower_is_better=False, on_train=False, name = 'valid_f1')


def train():
    wandb_run = wandb.init()
    config = wandb.config

    net = NeuralNetClassifier(
    ClassifierModule,
    max_epochs=10,
    lr=0.1,
    callbacks = [f1,WandbLogger(wandb_run)],
    train_split = skorch.dataset.CVSplit(5,stratified=True,random_state =42),
    device=device,  # uncomment this to train with CUDA
    )

    net.fit(X_train, y)

sweep_config = {
    'method': 'grid', # or 'grid' or 'random'
    'metric': {
        'name': 'accuracy',
        'goal': 'maximize'
    },
    'parameters':{
        'max_epochs': {
            'values':[10,15,20]
        }
    }
}

sweep_id = wandb.sweep(sweep_config,entity='utunex',project='test')
wandb.agent(sweep_id,train)
```
### Problem:
This code didn't display anything to wandb.

### Please help me on:
1. This is my first time to combine wandb and skorch and I do have no experience, please help check where the problem is in my code;  
2. I guess the problem may lie at the 'metric' item in sweep_config and I'm realy confused about how to set this item in skorch; the official documentation says:
> Specify the metric to optimize. This metric should be __logged explicitly to W&B by your training script__. using wandb.log({"val_loss" : valid_loss})  

This is where I feel confused. In the skorch part in my code, I used f1_score and give it a name as :valid_f1, so should I set thee metric as 'f1' or 'valid_f1'? and how to explicitly log it to wandb? I have used the wandblogger which 

> automatically log best trained model, all metrics from your net’s history, model topology and computer resources to Weights & Biases after each epoch. 

Isn't it enough?
In summary, I hope to know where is the mistake in my code and how to set the metric.

Thank you.

**Comment by raubitsj** [comment]
Hello, i tried to reproduce any issue and used your code exactly as you put it into the bug and I ran it (on a mac):

The W&B sweep url
https://app.wandb.ai/jeffr/bug_1212/sweeps/b4u5e8nf

And an excerpt from running the script:

<img width="903" alt="Screen Shot 2020-08-23 at 4 14 19 PM" src="https://user-images.githubusercontent.com/1832511/90991202-3e1c1080-e55c-11ea-917a-3e28cccf5db7.png">

If you can add information about your problem, that would be helpful.

What version of wandb are you using?
What OS?
Are you using a jupyter notebook?
Do you have any output on your screen that indicates that the script is not working?

Thanks

As for specifying the metric to maximize, that is only useful for bayes search or hyperband early termination.  In your case, if you want to maximize f1, you would specify a metric of "valid_f1"

**Comment by UTUnex** [comment]
Thank you for your reply and now I can add more information:
### More information  
wandb version:  0.9.5  
system: windows10 64 bit  
platform: Jupter notebook embedded in vscode  

### Problem  
After run this code :''wandb.agent(sweep_id,train)''
it displayed wandb: ''Agent Starting Run: yl5esssz with config:
	max_epochs: 10''
and keeps running but no more updates after that..
The notebook just displayed :''extcuting cell...'' but in the wandb, no more updates generated, for the current code scale, I dont think it needs that long time for training, so I think something must be wrong.
I don't know whether this is because of this issue:  

> Using wandb.agent() with jupyter notebook environments can __hang__ when using GPUs.
There can be a bad interaction between wandb.agent() and jupyter environments due to how GPU/CUDA resources are initialized by frameworks.
A temporary workaround (until we can fix these interactions) is to avoid using the python interface for running the agent.  Instead, use the command line interface by setting the program key in the sweep configuration, and execute: !wandb agent SWEEP_ID in your notebook.  

I guess this is the problem as until now, my code is just running but stays still at max epochs 10 without any further updates for epochs for 15 and 20, and also no more updates in wandb..

**Comment by UTUnex** [comment]
Can you offer an detailed example for wandb_sweep using in scikit learn? I didn't find any example in your github. As skorch has a sklearn API format, so I think a sklearn example will help.

**Comment by sydholl** [comment]
We are closing this issue due to inactivity. If you feel that it merits additional review, please provide a reproduction and add a comment explaining why. Thank you.
